int32 userID
touchpadInfo[] touchpadInfo
